food-bank-app/src/hooks/useClient/httpClient.ts

```diff
@@ -23,16 +29,50 @@ class HttpClient {
   }
 
   private readonly _initializeResponseInterceptor = (): void => {
-    this.instance.interceptors.response.use(this._handleResponse)
+    this.instance.interceptors.request.use(this._handleRequestAuthorization)
+    this.instance.interceptors.response.use(this._handleResponse, this._handleError)
+  }
+
+  private readonly _handleRequestAuthorization = async (initialConfig: AxiosRequestConfig): Promise<AxiosRequestConfig> => {
+    const requestConfig = initialConfig
+
+    const accessToken: string = await AsyncStorage.getItem('accessToken') ?? ''
+
+    if (requestConfig.headers?.Authorization == null && accessToken != null) {
+      requestConfig.headers = {
+        ...requestConfig.headers,
+        Authorization: `Bearer ${accessToken}`
+      }
+    }
+
```

Comment:
this basically adds the access token before every request, which will work for the rest of the calls after login... this is an async operation tho an I have not really tried it.

On web it is most easier as localStorage is synchronous... I've been thinking about using this package:
https://www.npmjs.com/package/sync-storage

Reply:
Have you checked [expo-secure-store]( https://docs.expo.dev/versions/latest/sdk/securestore/)? it's supposed to be encrypted and thus _safer_.